lib/paymentrails/gateways/RecipientGateway.rb

```diff
@@ -64,5 +83,19 @@ def recipient_list_builder(response)
       end
       recipients
     end
+
+    def payments_list_builder(response)
+      payments = []
+      data = JSON.parse(response)
+
+      data.each do |key, value|
+        next unless key === 'payments'
+        value.each do |newKey, _newValue|
+          payment = loosely_hydrate_model(Payment.new, newKey)
+          payments.push(payment)
+        end
+      end
+      payments
+    end
```

Comment:
this is a smell: A symptom of something wrong.

I think gateways and resource builders should be separate object so that this code could do something like `PaymentGatewayBuilder.build_list` instead of duping code

Reply:
@zero-bites I'd be okay to introduce this change if it makes the SDK better.
Let me know if introducing this will need us to update the scope of work.

I am inclined to introduce changes making SDK better and stable, and announce them in release notes.
If they introduce new public methods to developers, keep both old and new methods for time being and mark the old methods as Deprecated

Reply:
Implementing this would extend the scope of work as this change falls into the category of improving code architecture... Let's chat? I'll DM you

Reply:
Okay, let’s note this somewhere in a future-to-do list for now then, in favour of moving fast.